src/Game.js

```diff
@@ -30,47 +32,40 @@ class Game
 		this.gameState = 0;
 	}
 	
-	PlayGame()
+	SetRealizer(realizer)
 	{
-		CheckGameEnd();
-		while (turnIndex === 0)
-		{
-			DoTurn();
-			CheckGameEnd();
-		}
-		// TODO: Output winner / loser
+		this.realizer = realizer;
+		this.players.forEach((player) => {player.realizer = realizer; });
 	}
+
 	
-	DoTurn()
+	Step()
 	{
-		/* Get a legal move */
-		let proposedMove = this.nextTurn.GetMove();
-		while (!this.Validate(proposedMove)) proposedMove = this.nextTurn.GetMove();
-		
-		/* Make the move */
-		this.CommitMove(proposedMove);
-		
-		/* Get the next move */
-		this.lastTurn = this.nextTurn;
-		this.nextTurn = this.nextTurn.EndTurn();
-		if (this.nextTurn === undefined)
+		if (this.gameState === 0)
 		{
-			this.turnIndex = (this.turnIndex + 1) % this.turnOrder.length;
-			this.nextTurn = this.turnOrder[this.turnIndex];
+			this.DoTurn();
+			this.CheckGameEnd();
+		}
+		if (this.gameState !== 0)
```

Comment:
It's so that we can skip the first if when this is re-called by someone trying to enter another move, but so this if still runs immediately if the action taken ends the game.